app/controllers/ratings_controller.rb

```diff
@@ -0,0 +1,81 @@
+class RatingsController < ApplicationController
+  skip_before_action :verify_authenticity_token, :only => [:close, :update, :follow, :unfollow]
+
+  def index
+    params = unsafe_params_hash.select{|k, v| v.present? }
+
+    filter_params = {}
+
+    m_selected_category = Maybe(@current_community.categories.find_by_url_or_id(params[:category]))
+    filter_params[:categories] = m_selected_category.own_and_subcategory_ids.or_nil
+    selected_category = m_selected_category.or_nil
+    relevant_filters = select_relevant_filters(m_selected_category.own_and_subcategory_ids.or_nil)
+    relevant_search_fields = parse_relevant_search_fields(params, relevant_filters)
+
+    @categories = @current_community.categories.includes(:children)
+    @main_categories = @categories.select { |c| c.parent_id == nil }
+
+    @show_categories = @categories.size > 1
+    @category_menu_enabled = @show_categories || @show_custom_fields
+
+    @persons = Array.new()
+    Category::persons_by_listings_in_category(selected_category).flatten.uniq.each do |target_user|
```

Comment:
It is the only one place where you call `persons_by_listings_in_category` function. Maybe, it's better to move `flatten.uniq` logic to the function itself? Or do you have any plans to reuse it later?